lib/MusicBrainz/Server/Data/Event.pm

```diff
@@ -392,7 +392,7 @@ sub _find_performers
         $map->{$event_id} ||= [];
         push @{ $map->{$event_id} }, {
             entity => $artists->{$artist_id},
-            roles => $roles
+            roles => [ uniq map { $_ } @{ $roles } ]
```

Comment:
The `map` doesn't appear to be doing anything here. `uniq @{ $roles }` should be sufficient?

Reply:
True! (tbh I just fought the stuff for quite a while until I found something that worked and then I didn't dare to touch it again without supervision :) )